### ISSUE TITLE
get_objects_for_user raises TypeError for missing codenames in some cases

### ISSUE THREAD

**Issue by StenAkeStrid** [django-guardian/django-guardian#326]
I encountered an unexpected result when I misspelled a codename for the `get_objects_for_user` shortcut.

### How to reproduce ###
On current main branch devel#e9fdcb00a5b8151646637e9ae5094eb743f34e05
`get_objects_for_user` with klass specified, and one missing codename given, so not a full permission name.

### Expected result ###
It should probably raise DoesNotExist: ContentType matching query does not exist.

### Actual Result ###
It raises TypeError: unsupported operand type(s) for +: 'NoneType' and 'unicode'.
From row 416 in shortcuts.py in get_objects_for_user:
     
    if user.has_perm(app_label + '.' + code):

**Comment by mcgeeco** [comment]
Got this as well. Changing to 
```python 
if user.has_perm('%(app_label)s.%(code)s' % {'app_label': app_label, 'code': code}):
```
seems to fix it, but I haven't had a change to check yet.

**Comment by odedbd** [comment]
+1 
happened to me too after upgrading Guardian to 1.3 (Django 1.8.2). The suggested fix worked for me locally.

**Comment by bminortx** [comment]
@mcgeeco that worked out for me, too; my test suite _finally_ passes after the Django/Guardian double upgrade, haha. Nice fix.

**Comment by randlet** [comment]
Bumped into this also.  I just changed the line to:

```python
if app_label is not None and user.has_perm(app_label + '.' + code):
    global_perms.add(code)
```

which should be slightly more efficient than the fix from @mcgeeco since it won't hit the database looking for a permission like `None.app_code`.

**Comment by edufelipe** [comment]
:+1:  @lukaszb I think fixing this justifies cutting a 1.3.1 release, because it has broken a use case that was perfectly valid before.

**Comment by jscn** [comment]
++ for cutting a new release with this change.

**Comment by whitews** [comment]
Same here, just updated a project to Django 1.8 and guardian 1.3 and  ran into this issue. Any chance the fix can be merged and a new minor version released?

**Comment by GNewc** [comment]
Wondering why not still getting app_label off of ctype when klass is provided...? Adding the following allowed our code to function normally:
    if app_label is None:
      try:
        app_label = ctype.app_label
      except:
        import logging
        logging.getLogger(__name__).exception()
        app_label = 'None'

**Comment by brianmay** [comment]
This has been fixed in #334.
